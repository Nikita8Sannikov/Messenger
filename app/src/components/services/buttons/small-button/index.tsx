import ButtonComponent, { type IButtonComponent } from "@components/ui/button";

// Компонент маленькой кнопки
export default function SmallButtonComponent(props: Omit<IButtonComponent, "size">) {
	return <ButtonComponent {...props} size="small" />;
};