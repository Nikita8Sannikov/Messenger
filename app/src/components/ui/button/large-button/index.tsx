import ButtonComponent, { IButtonComponent } from "@components/ui/button";

export function LargeButtonComponent(props: Omit<IButtonComponent, "size">) {
    return <ButtonComponent {...props} size="large" />;
}