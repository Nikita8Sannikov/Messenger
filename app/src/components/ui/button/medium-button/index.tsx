import ButtonComponent, { IButtonComponent } from "@components/ui/button";

export function MediumButtonComponent(props: Omit<IButtonComponent, "size">) {
	return <ButtonComponent {...props} size="medium" />;
}