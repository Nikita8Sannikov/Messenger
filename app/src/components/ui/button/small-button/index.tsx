import ButtonComponent, { IButtonComponent } from "@components/ui/button";

export function SmallButtonComponent(props: Omit<IButtonComponent, "size">) {
    return <ButtonComponent {...props} size="small" />;
}