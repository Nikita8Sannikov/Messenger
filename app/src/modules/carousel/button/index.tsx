import { memo } from "react";
import { useTranslation } from "react-i18next";

import ButtonComponent from "@components/ui/button";
import ArrowRightIconComponent from "@components/icons/arrowRightIcon";
import ArrowLeftIconComponent from "@components/icons/arrowLeftIcon";

interface ICarouselButton {
    next?: boolean;
    isDisabled: boolean;
    handleClick: (dir: number) => void;
};

// Компонент, отвечающий за кнопки "Назад" и "Вперед" в карусели картинок
export default memo(function CarouselButton({ next, isDisabled, handleClick }: ICarouselButton) {
    const { t } = useTranslation();

    // Обработка кнопкок "Назад"/"Вперед"
    const onClick = () => {
        handleClick(next ? 1 : -1);
    }

    return <ButtonComponent size="small" onClick={onClick} disabled={isDisabled}>
        {next
            ? <>{ t("images-carousel-module.further") } <ArrowRightIconComponent size={26} /></>
            : <><ArrowLeftIconComponent size={26} /> { t("images-carousel-module.back") }</>
        }
    </ButtonComponent>
});