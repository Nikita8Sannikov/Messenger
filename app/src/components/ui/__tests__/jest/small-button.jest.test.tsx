import { screen, render } from "@testing-library/react";
import { SmallButtonComponent } from "@components/ui/button/small-button";


describe("SmallButton", () => {
    test("SmallButton with size prop", () => {
        render(<SmallButtonComponent >Test</SmallButtonComponent>);
        const boxElement = screen.getByText("Test")
        expect(boxElement)
    })

    test("matches snapshot", () => {
        const { asFragment } = render(<SmallButtonComponent>Test</SmallButtonComponent>);

        // Снимаем снапшот компонента
        expect(asFragment()).toMatchSnapshot();
    });
})