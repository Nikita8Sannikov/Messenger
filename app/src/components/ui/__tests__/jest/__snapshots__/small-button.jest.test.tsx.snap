// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallButton matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-1ccsd5i-MuiButtonBase-root-MuiButton-root"
    id=":r1:"
    tabindex="0"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;
